### ISSUE TITLE
do not run a comparison when you just unset the variable

### ISSUE THREAD

**Issue by darix** [openSUSE/post-build-checks#64]

**Comment by darix** [comment]
@JanZerebecki seems like this is your code originally.

**Comment by JanZerebecki** [comment]
yes, thx. I hope you don't mind if it takes me a while to get around to this again. I will need to touch this also because we found the idea won't work as is, so obs-build needs to be changed again.

**Comment by JanZerebecki** [comment]
Some other things I should consider when changing this: Also test with: osc shell

How to handle missing release during osc build: https://github.com/openSUSE/obs-build/pull/982#issuecomment-1954200264

There is some desire to make missing date not an error: https://bugzilla.suse.com/show_bug.cgi?id=1220540#c6

**Comment by JanZerebecki** [comment]
Also better wording so people are pointed to osc build --release

**Comment by darix** [comment]
using `--release` with `osc build` is something only for special occasions where you need to ensure it can be installed with `rpm|zypper --force`

it shouldnt be required for normal use.

**Comment by bnavigator** [comment]
Any news here? Using `osc build --release 1` is a good workaround to avoid misguided `python-bytecode-inconsistent-mtime` rpmlint errors for local builds. But it should not be necessary. I had to reject more than one SR from packagers you thought they had to adjust the mtime for obs builds.

**Comment by JanZerebecki** [comment]
I did not realize that possibility of interaction, thank you for reporting.

I did not get the chance, yet, to work on this again.

**Comment by JanZerebecki** [comment]
See also https://github.com/openSUSE/rpm-config-SUSE/pull/88

**Comment by JanZerebecki** [comment]
In https://bugzilla.suse.com/show_bug.cgi?id=1220540#c20 Bernhard said:

> After some discussion with mls we concluded that
> erroring out on a missing changelog date should be a separate option
> from setting SDE from changelog date.
> 
> This would allow devel projects to profit from reduced rebuilds
> while not having to follow Factory policy to force a .changes file.
> 
> proposed names for the new macro were:
> %_missing_source_date_epoch_terminate_build
> %_nonzero_exit_pkgcheck_terminate_build
